File: games/js/volume_bar.js
/*
 * Misho Georgiev game project
 *
 * 2022 Softel Labs
 *
*/


class VolumeBar {
  constructor(ctx, p1, max_vol, name, color, min_vol=0.0) {
    this.ctx = ctx
    this.w = 100
    this.h = 15
    this.p1 = p1
    this.max_vol = max_vol
    this.min_vol = min_vol
    this.name = name
    this.color = color

    this.p2 = [p1[0] + this.w, p1[1] + this.h]
  }

  draw(vol) {
    this.max_vol = Math.max(this.max_vol, vol)
    this.min_vol = Math.min(this.min_vol, vol)
    // vol = Math.min(this.max_vol, vol)
    // vol = Math.max(this.min_vol, vol)

    draw_text(this.ctx, this.name, this.p2[0] + 3, this.p2[1], 12, "white")
    draw_rect(this.ctx, this.p1[0], this.p1[1], this.w * vol / (this.max_vol - this.min_vol), this.h, this.color, true)
    draw_rect(this.ctx, this.p1[0], this.p1[1], this.w, this.h, "white")

    draw_text(this.ctx, parseInt(vol), this.p1[0] + this.w / 4, this.p1[1] + this.h / 2 + 3, 10, "white")
  }
}

File: games/py/volume_bar.py
import math
import cv2


class VolumeBar:
  def __init__(self, p1, max_vol, name, color):
    self.w = 100
    self.h = 10
    self.p1 = p1
    self.max_vol = max_vol
    self.name = name
    self.color = color

    self.p2 = [p1[0] + self.w, p1[1] + self.h]

  def __call__(self, frame, vol):
    self.draw(frame, vol)

  def draw(self, frame, vol):
    vol = min(self.max_vol, vol)
    p3 = [int(self.p1[0] + self.w * vol / self.max_vol), self.p1[1] + self.h]
    cv2.putText(frame, self.name, (self.p2[0] + 3, self.p2[1]), fontFace=cv2.FONT_HERSHEY_SIMPLEX, fontScale=0.5, color=(255, 255, 255), thickness=1)
    cv2.rectangle(frame, self.p1, p3, self.color, -1)
    cv2.rectangle(frame, self.p1, self.p2, (255, 255, 255), 1)
